esm/vs/language/json/json.worker.js

```diff
@@ -3583,17 +3585,65 @@ function validate(n, schema, validationResult, matchingSchemas) {
     }
   }
 }
+
+var dictionary = null;
+
+function getDictionary() {
+  if (!dictionary) {
+    dictionary = new Typo("en_US", false, false, { dictionaryPath: "/src/assets/static/typo-js/" });
+  }
+  return dictionary;
+}
+
 function parse3(textDocument, config) {
   var problems = [];
   var lastProblemOffset = -1;
   var text = textDocument.getText();
   var scanner = createScanner2(text, false);
   var commentRanges = config && config.collectComments ? [] : void 0;
   var multilineStringRanges = [];
+  var notSpelledCorrectly = [];
+
+  const dictionary = getDictionary();
+
   function _scanNext() {
     while (true) {
       var token_1 = scanner.scan();
       _checkScanError();
+      var currentWord = text.substring(scanner.getTokenOffset(), scanner.getTokenOffset() + scanner.getTokenLength());
+      if (/^".*"$/.test(currentWord)) {
+        var wordWithoutQuotes = currentWord.slice(1, -1);
+
+        var words = wordWithoutQuotes.match(/[a-zA-Z]+/g) || [];
+        var nonAlphaBefore = wordWithoutQuotes.match(/^[^a-zA-Z]+/) || [];
+        var delimiters = wordWithoutQuotes.match(/(?<=[a-zA-Z])[^a-zA-Z]+(?=[a-zA-Z])/g) || [];
+
+        var specialCharsBeforeLength = nonAlphaBefore.length > 0 ? nonAlphaBefore[0].length : 0;
+        
+        var cumulativeLength = specialCharsBeforeLength;
+
+        var spellResults = words.reduce((acc, word, index) => {
+          if (isNaN(word) && !dictionary.check(word)) {
+            var wordStartPosition = scanner.getTokenOffset() + cumulativeLength + 1;
+            var wordEndPosition = wordStartPosition + word.length
+
+            acc.push({
+              word,
+              suggestions: [],
+              start: wordStartPosition,
+              end: wordEndPosition
+            });
+          }
+
+          cumulativeLength += word.length + (delimiters.length > index ? delimiters[index].length : 0);
+          
+          return acc;
+        }, []);
```

Comment:
I think it's a bad practice to mutate from inside a reduce (we can pass the cumulativeLength as the part of the acc variable )